vector3d.h

```diff
@@ -0,0 +1,83 @@
+//3d vector class for Rendera, Yashas M Salian
+
+#include<math.h>
+
+class vec3
+{
+private:
+    float x, y, z;    //Components of vector are kept private
+public:
+    void xinput(float xin)    //functions used to input x, y, z
+    {
+        x = xin;
+    }   
+    
+    void yinput(float yin)
+    {
+        y = yin;
+    } 
+    
+    void zinput(float zin)
+    {
+        z = zin;
+    }
+    
+    float getx()    //functions used to obtain the stored values of x, y, z
+    {
+        return x;
+    }
+    
+    float gety()
+    {
+        return y;
+    }
+    
+    float getz()
+    {
+        return z;
+    }
+    
+    vec3 unitvect(vec3 vect)    //used to obtain unit vector in the direction of the vector vect
+    {
+        float mag;
+        vec3 unit;
+        mag = sqrt(vect.getx() * vect.getx() + vect.gety() * vect.gety() + vect.getz() * vect.getz());
+        unit.xinput(vect.getx() / mag);
+        unit.yinput(vect.gety() / mag);
+        unit.zinput(vect.getz() / mag);
+        return unit;
+    }
+    
+    vec3 operator + (vec3 vect)    //overloading + for vector addition
```

Comment:
It would also be nice to have `+=`, `-=`, `*=`, `/=`.

Ideally, what you want to do for `+`, `-`, `*`, and `/` is that you want to overload the respective compound assignment operators (aka `+=`, `-=`, `*=`, `/=`), and then define these ones in terms of those.

Doing this kind of bootstrapping reduces the amount of changes you have to make when you want to change some logic, and also makes the code a bit cleaner (in this case).

Reply:
Your `const`-correctness could be improved: The correct function signature for addition is `vec3 operator+(const vec3& vect)`. Refer to [this](https://gist.github.com/beached/38a4ae52fcadfab68cb6de05403fa393#operator-addition) for all the correct function signatures.

(Make sure to do this for all operators!)